kratos/includes/oriented_bounding_box.h

```diff
@@ -0,0 +1,472 @@
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief Computes the intersection between two OrientedBoundingBox (current and new)
+     * @details See https://www.geometrictools.com/Documentation/DynamicCollisionDetection.pdf
+     * @param rOtherOrientedBoundingBox The other Oriented Bounding Box considered
+     */
+    bool DirectHasIntersection(const OrientedBoundingBox& rOtherOrientedBoundingBox) const;
+
+    /**
+     * @brief Computes the intersection between two OrientedBoundingBox (current and new)
+     * @details See: https://gamedev.stackexchange.com/questions/112883/simple-3d-obb-collision-directx9-c
+     * @param rOtherOrientedBoundingBox The other Oriented Bounding Box considered
+     */
+    bool SeparatingAxisTheoremHasIntersection(const OrientedBoundingBox& rOtherOrientedBoundingBox) const;
+
+    /**
+     * @brief Check if there's a separating plane in between the selected axis
+     * @param rRelativePosition The difference between the center of the both boxes
+     * @param rPlane The vector which defines the plane
+     * @param rOtherOrientedBoundingBox The other Oriented Bounding Box considered
+     */
+    bool GetSeparatingPlane(
+        const array_1d<double, 3>& rRelativePosition,
+        const array_1d<double, 3>& rPlane,
+        const OrientedBoundingBox& rOtherOrientedBoundingBox
+        ) const;
+
+    /**
+     * @brief Check if there's a separating plane in between the selected axis (2D version)
+     * @param rRelativePosition The difference between the center of the both boxes
+     * @param rPlane The vector which defines the plane
+     * @param rOtherOrientedBoundingBox The other Oriented Bounding Box considered
+     */
+    bool GetSeparatingPlane2D(
+        const array_1d<double, 3>& rRelativePosition,
+        const array_1d<double, 3>& rPlane,
+        const OrientedBoundingBox& rOtherOrientedBoundingBox
+        ) const;
+
+    /**
+     * @brief Check if there's a separating plane in between the selected axis (3D version)
+     * @param rRelativePosition The difference between the center of the both boxes
+     * @param rPlane The vector which defines the plane
+     * @param rOtherOrientedBoundingBox The other Oriented Bounding Box considered
+     */
+    bool GetSeparatingPlane3D(
+        const array_1d<double, 3>& rRelativePosition,
+        const array_1d<double, 3>& rPlane,
+        const OrientedBoundingBox& rOtherOrientedBoundingBox
+        ) const;
+
+    /**
+     * @brief Fast compute of cross product without considering checks. Performs the cross product of the two input vectors a,b
+     * @details a,b are assumed to be of size 3
+     * @param a First input vector
+     * @param b Second input vector
+     * @return c The resulting vector
+     */
+    static array_1d<double, 3> FastCrossProduct(const array_1d<double, 3>& a, const array_1d<double, 3>& b)
```

Comment:
Why we need this method? Shouldn't be in math utils?

Reply:
Could be

Reply:
The only difference is that t skips the checks

Reply:
The tests are in debug only. So please use the math utils one

Reply:
OK